### PR TITLE
Remove obsolet files and log file symlink code

### PR DESCRIPTION
**Pull request by yubiuser**
**What does this PR aim to accomplish?:**

Removes obsolet files in v6 and the code creating the log file symlinks in `/var/log/` which was a temporal fix for not breaking users scripts. It was introduced two years ago with https://github.com/pi-hole/pi-hole/pull/3732 moving all logfiles to `/var/log/pihole`

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

**Comment by rdwebdesign** [comment]
This is also marked to be removed in v6:

https://github.com/pi-hole/pi-hole/blob/15d7fab2cbfe6e80eec0b776a222281126bc79d4/advanced/Templates/pihole-FTL-prestart.sh#L37-L46

**Comment by yubiuser** [comment]
Done.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -2323,34 +2323,6 @@ main() {
...
-    # we need to move any existing `pihole*` logs from `/var/log` to `/var/log/pihole`
-    # if /var/log/pihole.log is not a symlink (set during FTL startup) move the files
-    # can be removed with Pi-hole v6.0
-    # To be sure FTL is not running when we move the files we explicitly stop it here
-
-    stop_service pihole-FTL &>/dev/null
-
-    if [ ! -d /var/log/pihole/ ]; then
-        mkdir -m 0755 /var/log/pihole/
-    fi
```
Comment (DL6ER) [inline automated install/basic-install.sh:2336]: Do we need to keep this? I have not found any other location where we are creating the directory (e.g. for fresh installs).

Comment (yubiuser) [reply automated install/basic-install.sh:2336]: Look here: https://github.com/pi-hole/pi-hole/blob/67de7be802d942947dc1d401de7527e3f8605923/advanced/Templates/pihole-FTL-prestart.sh#L14

Comment (DL6ER) [reply automated install/basic-install.sh:2336]: Ah, yeah, my mistake. I `grep`ped for `/var/log/pihole/` including a trailing slash which didn't reveal this one.

**Review by DL6ER — APPROVED** [review]

**Pull request merged**